nimlangserver.nim

```diff
@@ -652,11 +657,14 @@ proc rename(ls: LanguageServer, params: RenameParams, id: int): Future[Workspace
     position: params.position
   ))
   # Build up list of edits that the client needs to perform for each file
+  let projectDir = ls.openFiles[params.textDocument.uri].projectFile.await().parentDir()
   var edits = newJObject()
   for reference in references:
-    if reference.uri notin edits:
-      edits[reference.uri] = newJArray()
-    edits[reference.uri] &= %TextEdit(range: reference.range, newText: params.newName)
+    # Only rename symbols inside the project (Stops modifying the stdlib, external packages)
+    if reference.uri.toPath().isRelativeTo(projectDir):
+      if reference.uri notin edits:
```

Comment:
I would expect to return an error if we cannot rename a particular symbol. In general, there is a method prepareRename that can be used to validate if the symbol at the point can be renamed. We can also use def call in prepareRename to check if there is something to be renamed at the point.